src/server.c

```diff
@@ -5977,6 +5978,7 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
                 "master_host:%s\r\n", server.masterhost,
                 "master_port:%d\r\n", server.masterport,
                 "master_link_status:%s\r\n", (server.repl_state == REPL_STATE_CONNECTED) ? "up" : "down",
+                "master_connect_time:%jd\r\n", (intmax_t)server.repl_master_connect_time,
                 "master_last_io_seconds_ago:%d\r\n", server.master ? ((int)(server.unixtime-server.master->lastinteraction)) : -1,
```

Comment:
Almost timestamps in INFO are relative to the current time, including `master_last_io_seconds_ago` and `master_link_down_since_seconds`. These also contain the words "seconds" and "ago/since" which make them extra clear about what they represent.

For symmetry with those, maybe we should call this new one `master_link_up_since_seconds` and only display it if we're up?

Reply:
Btw, are we interested in how long the reconnect took? I.e. how long we the connection was down before it came up?

Reply:
+1 about including seconds.

Reply:
That's what I meant, like the existing `master_link_down_since_seconds` but for up instead of down.

Reply:
At first, I also considered using something like "master_connection_uptime_seconds", but as you mentioned, if the replica keep retrying, this value will remain 0, which is not conducive to observation. On the other hand, `master_connect_time` is a monotonically increasing timestamp, and its monitoring trend can easily detect issues (as it increases with each new synchronization or retry).

In addition, `master_connect_time` and `master_link_up_since_seconds` are not exactly the same. `master_connect_time` focuses on the "connect" action and its underlying causes (such as a synchronization connection being disconnected), while `master_link_up_since_seconds` represents the time when a complete and healthy recovery occurs. Of course, they are not mutually exclusive, and we can have both metrics.

Reply:
Ah, even while down you want to see when the connect attempt started. I would add "attempt" to the name to make it more clear.